py/helpers/cache/cache_helper.py

```
@@ -1,64 +1,109 @@
+"""Cache Helper Module."""
+
 import hashlib
 import json
 
 from .cache_factory import CacheFactory
 from .cache_factory import RedisCache
 
 
-def sort_nested_dict_or_list(d):
-  if isinstance(d, dict):
-    return {k: sort_nested_dict_or_list(v) for k, v in sorted(d.items())}
-  elif isinstance(d, list):
-    return [sort_nested_dict_or_list(x) for x in d]
+def sort_nested_dict_or_list(data: dict | list) -> dict | list:
+  """Sort the nested dict or list
+
+  Args:
+      data (dict | list): input dict or list
+
+  Returns:
+      dict | list: sorted dict or list
+  """
+  if isinstance(data, dict):
+    return {k: sort_nested_dict_or_list(v) for k, v in sorted(data.items())}
+  elif isinstance(data, list):
+    return [sort_nested_dict_or_list(x) for x in data]
   else:
-    return d
+    return data
```

raise error or at least log the warning if the data is of unexpected format (instead of silently returning)

---

return data for the recursive code , and yes adding logs for unexpected datatypes other than dict or list for debug   purpose